compiler/semantic/sql.go

```diff
@@ -322,35 +325,141 @@ func (a *analyzer) semSelectValue(sel *ast.Select, sch schema, seq dag.Seq) (dag
 	return seq, &dynamicSchema{}
 }
 
+func (a *analyzer) semValues(values *ast.Values, seq dag.Seq) (dag.Seq, schema) {
+	var schema *super.TypeRecord
+	exprs := make([]dag.Expr, 0, len(values.Exprs))
+	sctx := super.NewContext()
+	for _, astExpr := range values.Exprs {
+		e := a.semExpr(astExpr)
+		val, err := kernel.EvalAtCompileTime(sctx, e)
+		if err != nil {
+			a.error(astExpr, errors.New("expressions in values clause must be constant"))
+			return seq, badSchema()
+		}
+		recType, ok := super.TypeUnder(val.Type()).(*super.TypeRecord)
+		if !ok {
+			// Turn non record values into single column tuple
+			f := &dag.Field{Kind: "Field", Name: "c0", Value: e}
+			e = &dag.RecordExpr{Kind: "RecordExpr", Elems: []dag.RecordElem{f}}
+			val, _ := kernel.EvalAtCompileTime(sctx, e)
+			recType = super.TypeUnder(val.Type()).(*super.TypeRecord)
+		}
+		if schema == nil {
+			schema = recType
+		} else if schema != recType {
+			a.error(astExpr, errors.New("values clause must contain uniformly typed values"))
+		}
+		exprs = append(exprs, e)
+	}
+	if schema == nil {
+		a.error(values, errors.New("values clause must contain uniformly typed values"))
+		return seq, badSchema()
+	}
+	columns := make([]string, 0, len(schema.Fields))
+	for _, f := range schema.Fields {
+		columns = append(columns, f.Name)
+	}
+	seq = append(seq, &dag.Yield{
+		Kind:  "Yield",
+		Exprs: exprs,
+	})
+	return seq, &anonSchema{columns}
+}
+
 func (a *analyzer) genDistinct(e dag.Expr, seq dag.Seq) dag.Seq {
 	return append(seq, &dag.Distinct{
 		Kind: "Distinct",
 		Expr: e,
 	})
 }
 
-func (a *analyzer) semSQLPipe(op *ast.SQLPipe, seq dag.Seq, alias string) (dag.Seq, schema) {
+func (a *analyzer) semSQLPipe(op *ast.SQLPipe, seq dag.Seq, alias *ast.TableAlias) (dag.Seq, schema) {
 	if len(op.Ops) == 1 && isSQLOp(op.Ops[0]) {
 		seq, sch := a.semSQLOp(op.Ops[0], seq)
-		return derefSchema(sch, alias, seq)
+		outSeq, outSch, err := derefSchemaWithAlias(sch, alias, seq)
+		if err != nil {
+			a.error(op.Ops[0], err)
+		}
+		return outSeq, outSch
 	}
 	if len(seq) > 0 {
 		panic("semSQLOp: SQL pipes can't have parents")
 	}
-	return a.semSeq(op.Ops), &dynamicSchema{name: alias}
+	var name string
+	if alias != nil {
+		name = alias.Name
+		if len(alias.Columns) != 0 {
+			a.error(alias, errors.New("cannot apply aliased columns to dynamically typed data"))
```

Comment:
Nit: Use the same string as below.
```suggestion
			a.error(alias, errors.New("cannot apply column aliases to dynamically typed data"))
```